docs/CHANGELOG.md

```diff
@@ -13,6 +13,29 @@ Changes are grouped as follows
 - `Fixed` for any bug fixes.
 - `Security` in case of vulnerabilities.
 
+## [0.99.25] - 07-06-24
+### Added
+- `create_external_id_factory` in `external_id_factories` which takes in a configuration to produce external_id's of the
+  format `prefix:suffix`
+- `ExternalIdFactory` class as a template for custom functions to define for `DataModelWrite.external_id_factory`
+- Helper functions for unique external_id generation
+  - `domain_name` returns a clean string from the domain class name
+  - `uuid` returns a uuid
+  - `sha256` returns a sha256 has of the data
+  - `incremental_id` returns an incremental integer based on domain class
+- `incremental_id_factory` an instance of the `ExternalIdFactory` that implements the `incremental_id` helper
+
+### Changed
+- More flexible external_id_factory functions
+  - `sha256_factory` is now an instance of the `ExternalIdFactory` class
+  - `uuid_factory` is now an instance of the `ExternalIdFactory` class
+
+### Removed
+- The following functions in `external_id_factories` have been removed and replaced
+  - `create_incremental_factory` is replaced by `create_external_id_factory(suffix_ext_id_factory=incremental_id_factory)`
+  - `create_sha256_factory` is replaced by `create_external_id_factory(suffix_ext_id_factory=sha256_factory)`
+  - `create_uuid_factory` is replaced by `create_external_id_factory()`
```

Comment:
Then the changelog entry needs to be updated as well, they are no longer removed, instead deprecated.